features/activity/cancel_try_cancel/feature.go

```diff
@@ -0,0 +1,79 @@
+package cancel_try_cancel
+
+import (
+	"context"
+	"time"
+
+	"go.temporal.io/sdk-features/harness/go/harness"
+	"go.temporal.io/sdk/activity"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/temporal"
+	"go.temporal.io/sdk/workflow"
+)
+
+var Feature = harness.Feature{
+	Workflows:  Workflow,
+	Activities: activities,
+	// Effectively disable heartbeat throttling
+	// TODO(cretz): Pending https://github.com/temporalio/sdk-go/issues/859
+	// WorkerOptions: worker.Options{MaxHeartbeatThrottleInterval: 1 * time.Nanosecond},
+	Execute: func(ctx context.Context, runner *harness.Runner) (client.WorkflowRun, error) {
+		// Put client on activities
+		activities.Client = runner.Client
+		return runner.ExecuteDefault(ctx)
+	},
+}
+
+func Workflow(ctx workflow.Context) error {
+	// Start an activity
+	actCtx, actCancel := workflow.WithCancel(workflow.WithActivityOptions(ctx, workflow.ActivityOptions{
+		ScheduleToCloseTimeout: 1 * time.Minute,
+		HeartbeatTimeout:       5 * time.Second,
+		RetryPolicy:            harness.RetryDisabled,
+		// This is the default, just setting it here to be clear
+		WaitForCancellation: false,
+	}))
+	fut := workflow.ExecuteActivity(actCtx, activities.CancellableActivity)
+
+	// Sleep for the smallest amount of time (force task turnover)
+	if err := workflow.Sleep(ctx, 1*time.Millisecond); err != nil {
+		return err
+	}
+
+	// Cancel and confirm the activity errors with the cancel
+	actCancel()
+	if err := fut.Get(ctx, nil); !temporal.IsCanceledError(err) {
+		return harness.AppErrorf("expected activity cancel error, got: %v", err)
+	}
```

Comment:
No timeout here? There's a timeout in all the language implementations.

Reply:
I can add one, though it's not used as much in Go as it is in other languages and a test running forever is just as much of a non-success as a test that throws a timeout.

Reply:
Removed timeout on all of them